mewarpx/mewarpx/appendablearray.py

```diff
@@ -0,0 +1,242 @@
+        self.cleardata()
+        self.append(data)
+
+    def compressdata(self, data, whereidx):
+        """
+        Takes in data and copies into preallocated array using boolean indexing
+
+        Input:
+            data (array):  raw data to be copied
+            whereidx (boolean array): must be same size as first index in data
+        """
+
+        # Make sure the input data is an array and the dimensions match the unitshape
+        if not isinstance(data, numpy.ndarray):
+            raise Exception("data: must be a numpy array")
+
+        # Make sure where is a boolean array
+        if ((not isinstance(whereidx, numpy.ndarray))
+            or (whereidx.dtype != bool)
+            or (len(whereidx) != data.shape[0])):
+            raise Exception("whereidx must be a numpy array of booleans, with"\
+                            + " the same size as the first dimension of data")
+
+        self.cleardata()
+
+        # Make sure there is enough room in the array
+        datalen = whereidx.sum()
+        self._extend(datalen)
+        self._datalen = datalen
+
+        # Copy compressed data into array
+        outview = self._array[:self._datalen,...]
+        numpy.compress(whereidx, data, axis=0, out=outview)
+
+    def takedata(self, data, idx):
+        """
+        Takes in data and copies into preallocated array using fancy indexing
+
+        Input:
+            data (array):  raw data to be copied
+            idx (int array): must be equal or less in size as first index in data
+        """
+
+        # Make sure the input data is an array and the dimensions match the unitshape
+        if not isinstance(data, numpy.ndarray):
+            raise Exception("data: must be a numpy array")
+
+        # Make sure where is an int array
+        if ((not isinstance(idx, numpy.ndarray))
+            or (idx.dtype != int)):
+            raise Exception("idx must be a numpy array of ints")
+
+        self.cleardata()
+
+        # Make sure there is enough room in the array
+        datalen = len(idx)
+        self._extend(datalen)
+        self._datalen = datalen
+
+        # Copy compressed data into array
+        outview = self._array[:self._datalen,...]
+        numpy.take(data, idx, axis=0, out=outview)
+
+
+    def unitshape(self):
+        if self._unitshape is None:
+            return (1,)
+        else:
+            return self._unitshape
+
+    def reshape(self,newunitshape):
+        """
+    Change the shape of the appendable unit. Can only be used if a unitshape was
+    specified on creation.
+     - newunitshape: must have the same number of dimensions as the original
+                     unitshape
+        """
+        assert self._unitshape is not None,\
+               'Only an array with a specified unitshape can be reshaped'
+        assert len(newunitshape) == len(self._unitshape),\
+               ('New unitshape must have the same number of dimensions as original'+
+                'unitshape')
```

Comment:
```suggestion
               ('New unitshape must have the same number of dimensions as original '+
                'unitshape')
```

Reply:
Nice catch. The appendable array is coming from PR #47, please add this comment there.